src/toil/provisioners/gceProvisioner.py

```diff
@@ -368,7 +368,7 @@ def worker(driver, instance):
 
         threads = []
         for instance in instances:
-            t = threading.Thread(target=worker, args=(self._gceDriver,instance))
+            t = threading.Thread(target=worker, args=(self._gceDriver,instance), daemon=True)
```

Comment:
These threads are also one-shot threads, and it's somewhat important for them to execute as much as they can, because they delete cloud instances that cost money. I don't think they should be daemons.

Really this should be a thread pool; who knows when we'll have more instances to shut down than we have thread ulimit available.

Reply:
I went to turn this into a thread pool, but we don't actually have this under CI or any Google money, so I'd rather not refactor it without being able to test it.